docs/_docs/04_reference/02-cli-options.md

```diff
@@ -2208,8 +2208,10 @@ block is not evaluated until _after_ the queue has been populated with units to
 
 When using many dependencies, this option can speed up the dependency processing by fetching dependency output directly
 from the state file instead of using `tofu/terraform output` to fetch them.
-NOTE: This is an experimental feature, use with caution.
-Currently only AWS S3 backend is supported.
+
+**NOTE**: This is an experimental feature, use with caution. There is no guarantee that OpenTofu/Terraform will maintain the existing schema of their state files, so there is also no guarantee that the flag will work as expected in future versions of OpenTofu/Terraform. Also, note that the only backend that supports this feature is the AWS S3 backend.
```

Comment:
Maybe the suggestion is, rather than "use with caution" it's "Use with pinned versions of OpenTofu/Terraform, and be on the lookout when upgrading that if the tools change the underlying scheme, Terragrunt will also require updates and there may be version incompatibilities for a period of time" or similar.

Reply:
All that said -- has the s3 state file format changed in years?  Is this really something our users need to be concerned about, or is it more of a "true, but very unlikely to be a problem" kinda thing?

Reply:
AFAIK, it's never changed, and we're coordinating with OpenTofu maintainers to avoid any changes there if possible, but we have no influence on Terraform development.

Reply:
Added some notes.